src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs

```diff
@@ -38,6 +49,37 @@ public ModalImporterPresenter(IModalImporterView view, IDataSetToDataRepositoryM
                {
                   var dataRepo = _dataRepositoryMapper.ConvertImportDataSet(d.DataSource, i++, pair.Key);
                   dataRepo.ConfigurationId = id;
+                  var molecule = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation);
+                  var moleculeDescription = (metaDataCategories?.FirstOrDefault(md => md.Name == dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)?.ListOfValues.FirstOrDefault(v => v.Key == dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)))?.Value;
+                  var molecularWeightDescription = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation);
+
+
+                  if (moleculeDescription != null)
+                  {
+                     if (string.IsNullOrEmpty(molecularWeightDescription))
+                     {
+                        molecularWeightDescription = moleculeDescription;
+                        if (!string.IsNullOrEmpty(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation))
+                        dataRepo.ExtendedProperties.Add(new ExtendedProperty<string>() { Name = dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation, Value = moleculeDescription });
+                     }
+                     else
+                     {
+                        double molWeight;
+                        double moleculeMolWeight;
+                        double.TryParse(moleculeDescription, out moleculeMolWeight);
+                        double.TryParse(molecularWeightDescription, out molWeight);
+                        if (Math.Abs(moleculeMolWeight - molWeight) > Constants.DOUBLE_RELATIVE_EPSILON)
```

Comment:
This might be too low of a tolerance no?

Reply:
You are right, this is just a DTO. I renamed ImageComboBoxOption to RowOptionDTO. 

@msevestre @Yuri05: Constants.DOUBLE_RELATIVE_EPSILON is 1e-5. Four decimal digits might be restrictive indeed but it sounds also like the standard precision that I remember from statistics lessons at school. I guess it will depend on the standards you have defined already. Please, tell me the exact tolerance you have in mind and I will set it to the value proposed.

Reply:
1e-5 sounds good (is also what I use for comparisons in the most cases)

Reply:
My bad. I thought this was the .NET double tolerance.
WE should have somewhere a method that compares two doubles and returns of they are equal or not (might just be in PKSim). Let me check